runtime/src/main/java/org/corfudb/runtime/view/stream/AbstractQueuedStreamView.java

```diff
@@ -259,8 +264,13 @@ protected ILogData read(final long address, long readStartTime) {
     protected List<ILogData> readAll(@Nonnull List<Long> addresses) {
         try {
             Map<Long, ILogData> dataMap =
-                    runtime.getAddressSpaceView().read(addresses);
-            return addresses.stream().map(dataMap::get).collect(Collectors.toList());
+                    runtime.getAddressSpaceView().read(addresses, options.ignoreTrimmed);
+            // If trimmed exceptions are ignored, the data retrieved by the read API might not correspond
+            // to all requested addresses, for this reason we must filter out data entries not included (null).
+            // Also, we need to preserver ordering for checkpoint logic.
+            return  addresses.stream().map(dataMap::get)
+                    .filter(data -> data != null)
```

Comment:
I think a cleaner way to do this is to change read() API,  the returned map of read() an ordered map, then you can just return the ordered value set of the `dataMap`

Reply:
Fixed.